sensor/common/component.go

```diff
@@ -20,12 +20,17 @@ const (
 	SensorComponentEventOfflineMode SensorComponentEvent = "offline-mode"
 )
 
+// Notifiable is the interface used by Sensor to notify components of state changes in Central<->Sensor connectivity.
+type Notifiable interface {
+	Notify(e SensorComponentEvent)
```

Comment:
I think 

```
type EventHandler interface {
       Handle(e SensorComponentEvent)
}
```

makes more sense, but that'll cause more code changes, and may be annoying to do

Reply:
I think event handler might be a bit too generic and too similar to handling central events. The `Notify` name came from the need of notifying all components of state changes. Perhaps the name that is a bit misleading here is `SensorComponentEvent`, which could be `SensorState` or something similar.

Reply:
yeah this is hard to name. I originally thought `Notifier`, but it's not necessarily notifying anything, but rather it is handling notifications